### ISSUE TITLE
oxfmt: collapses space after closing backtick when code span contains escaped backtick

### ISSUE THREAD

**Issue by ftzi** [oxc-project/oxc#24966]
## Bug

`oxfmt` (markdown mode) collapses the space between an inline code span and the following word when the code span contains an escaped backtick (`` \` ``) and is followed by another code span on the same line.

## Reproduction

```sh
mkdir /tmp/oxfmt-repro && cd $_
echo 'a `b\`c` d `e` f.' > repro.md
bunx oxfmt@0.60.0 repro.md
cat repro.md
```

## Expected

```md
a `b\`c` d `e` f.
```

## Actual

```md
a `b\`c`d`e` f.
```

The space after the first code span (and the space after the second code span) is silently removed.

## Minimal trigger pattern

The bug requires both:

1. An inline code span containing an escaped backtick — `` `something-with-\`-backtick` ``
2. A space, then another code span (or text) on the same line

### Triggers

- `` `b\`c` d `e` f. `` → `` `b\`c`d`e` f. ``
- `` `b\`c` d `e\`f` g. `` → `` `b\`c`d`e\`f` g. ``

### Does NOT trigger

- `` `b\`c` d e f. `` (no following code span)
- `` `b\`c`next text. `` (no space between the two code spans)
- `` `b` c `d` e. `` (no escaped backtick in first span)
- `` `b` c `d\`e` f. `` (escape only in the *second* span, not the first)

## Environment

- oxfmt 0.60.0
- Markdown mode (default)
- Linux / macOS — both reproduce
- No config needed; default settings reproduce

## Workaround

Reword the markdown to avoid the trigger pattern. Either drop the escaped backticks in the first code span or move the example into a fenced code block.

## Context

Hit in a skill file where the example needed to literally show escaped backticks inside an inline code reference. Prettier (the reference implementation) preserves the space correctly.

**Comment by leaysgur** [comment]
Currently, Oxfmt simply delegates the formatting of Markdown files to Prettier, so this is not something we can change on our side.

---

> Prettier (the reference implementation) preserves the space correctly.

Prettier (v3.9.6) produces exactly the same output.

---

That said, I don't think this is a bug in the first place:
Per CommonMark, backslash escapes do not work inside code spans, so `` `b\`c` `` parses as a code span `` `b\` `` followed by literal text and the d in between is itself a code span whose surrounding spaces are stripped.

https://spec.commonmark.org/0.31.2/#example-338

To literally include a backtick in a code span, use a longer backtick
string as the delimiter (e.g. ``` ``b`c`` ```) instead of a backslash.
